controllers/storagecluster/noobaa_system_reconciler.go

```diff
@@ -193,6 +193,14 @@ func (r *StorageClusterReconciler) setNooBaaDesiredState(nb *nbv1.NooBaa, sc *oc
 		nb.Spec.DisableLoadBalancerService = sc.Spec.MultiCloudGateway.DisableLoadBalancerService
 	}
 
+	if sc.Spec.MultiCloudGateway != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig != nil && sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName != "" {
+		nb.Spec.ExternalPgSSLRequired = true
+		nb.Spec.ExternalPgSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretName, Namespace: sc.Spec.MultiCloudGateway.ExternalPgConfig.PGSecretNamespace}
+		nb.Spec.ExternalPgSSLUnauthorized = sc.Spec.MultiCloudGateway.ExternalPgConfig.AllowSelfSignedCerts
+		if sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName != "" && sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretNamespace != "" {
+			nb.Spec.ExternalPgSSLSecret = &corev1.SecretReference{Name: sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretName, Namespace: sc.Spec.MultiCloudGateway.ExternalPgConfig.TLSSecretNamespace}
```

Comment:
Do we need a check here to ensure that both name and namespace are set not either one of them?

Reply:
It will work with either one but just in case added both to for secret ref.